### PR TITLE
Migrate existing integration tests to a single binary

### PR DESCRIPTION
**Pull request by BrynCooke**
Each integration tests takes time to link, so this will help with build times.
It also makes things a bit less chaotic as we can have modules for integration tests.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-router/tests/telemetry_test.rs [hunk 1]
```diff
@@ -1,55 +0,0 @@
...
-use std::result::Result;
-
-use tower::BoxError;
-
-use crate::common::IntegrationTest;
-use crate::common::Telemetry;
-
-#[ignore]
-#[tokio::test(flavor = "multi_thread")]
-async fn test_otlp_tracing() -> Result<(), BoxError> {
```
Comment (SimonSapin) [inline apollo-router/tests/telemetry_test.rs:13]: Is removing rather than moving these deliberate?

**Review by SimonSapin — APPROVED** [review]
Do we also want [`autotests = false`](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#target-auto-discovery) in `Cargo.toml`?

File: apollo-router/tests/telemetry_test.rs (hunk 1 above)
Comment (BrynCooke) [reply apollo-router/tests/telemetry_test.rs:13]: Yeah, these tests are ignored anyway, and they didn't have anything actually set up to verify. They were just placeholders.

**Comment by BrynCooke** [comment]
> Do we also want [`autotests = false`](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#target-auto-discovery) in `Cargo.toml`?

I'm not sure. It's not as if there are extra binaries being created because of bin directories. What's the reason for wanting this?

**Comment by SimonSapin** [comment]
By default (`autotests = true`), every `tests/*.rs` file gets compiled as an integration test executable. From the diff it’s not clear if after this PR there is more than one such file remaining. Setting `autotests = false` would allow having such files (such as to use a modules from the "main" one) but not have them compiled as a separate executable. (Only explicit `[[test]]` TOML sections would create an integration test)

**Comment by BrynCooke** [comment]
| By default (autotests = true), every tests/*.rs file gets compiled as an integration test executable. From the diff it’s not clear if after this PR there is more than one such file remaining. Setting autotests = false would allow having such files (such as to use a modules from the "main" one) but not have them compiled as a separate executable. (Only explicit [[test]] TOML sections would create an integration test)

We're going to have to do this separately. I had a quick go, and the act of doing this for the existing tests changes the mod hierarchy.
Maybe there is some way around this.

**Pull request merged**